src/main/java/org/jlab/myquery/MySamplerWebService.java

```diff
@@ -40,6 +43,19 @@ public <T extends Event> EventStream<T> openEventStream(Metadata<T> metadata, In
         Instant end = begin.plusMillis(intervalMillis * (sampleCount - 1));
         EventStream<T> stream = nexus.openEventStream(metadata, begin, end, DataNexus.IntervalQueryFetchStrategy.STREAM, updatesOnly);
 
-        return new MySamplerStream<>(stream, begin, intervalMillis, sampleCount, priorEvent, updatesOnly, metadata.getType());
+        MySamplerStream<T> out;
+        try {
+            out = MySamplerStream.getMySamplerStream(stream, begin, intervalMillis, sampleCount, priorEvent, updatesOnly, metadata.getType());
+        } catch (Exception ex) {
+            if (stream != null) {
+                try {
+                    stream.close();
+                } catch (IOException closeIssue) {
+                    LOGGER.log(Level.SEVERE, "Could not close stream", ex);
+                }
+            }
+            throw ex;
```

Comment:
Catch and re-throw caught my eye, but I think is OK.   In case when exception is thrown internal cleanup is needed.